src/meta/src/hummock/manager/compaction/compaction_group_schedule.rs

```diff
@@ -711,3 +546,343 @@ impl HummockManager {
+                            tracing::error!(
+                                error = %e.as_report(),
+                                "failed to split_huge_compaction_group table {:?} from group-{}",
+                                table_ids,
+                                group.group_id
+                            );
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    pub async fn try_merge_compaction_group(
+        &self,
+        table_write_throughput: &HashMap<u32, VecDeque<u64>>,
+        group: &TableGroupInfo,
+        group_config: CompactionGroup,
+        next_group: &TableGroupInfo,
+        _next_group_config: CompactionGroup,
+        checkpoint_secs: u64,
+        created_tables: &HashSet<u32>,
+    ) -> Result<()> {
+        // TODO: remove this check after refactor group id
+        if group.group_id == StaticCompactionGroupId::StateDefault as u64
+            && next_group.group_id == StaticCompactionGroupId::MaterializedView as u64
+        {
+            return Ok(());
+        }
+
+        if group.table_statistic.is_empty() || next_group.table_statistic.is_empty() {
+            return Err(Error::CompactionGroup(format!(
+                "group-{} or group-{} is empty",
+                group.group_id, next_group.group_id
+            )));
+        }
+
+        // do not merge the compaction group which is creating
+        if check_is_creating_compaction_group(group, created_tables) {
+            return Err(Error::CompactionGroup(format!(
+                "Not Merge creating group {} next_group {}",
+                group.group_id, next_group.group_id
+            )));
+        }
+
+        // do not merge high throughput group
+        let window_size =
+            self.env.opts.merge_group_statistic_window_times / (checkpoint_secs as usize);
+
+        // merge the group which is low write throughput
+        if !check_is_low_write_throughput_compaction_group(
+            table_write_throughput,
+            checkpoint_secs,
+            window_size,
+            self.env.opts.table_low_write_throughput_threshold,
+            group,
+            self.env.opts.table_statistic_low_write_throughput_ratio,
+        ) {
+            return Err(Error::CompactionGroup(format!(
+                "Not Merge high throughput group {} next_group {}",
+                group.group_id, next_group.group_id
+            )));
+        }
+
+        let size_limit = (group_config.max_estimated_group_size() as f64
+            * self.env.opts.compaction_group_size_threshold) as u64;
+
+        if (group.group_size + next_group.group_size) > size_limit {
+            return Err(Error::CompactionGroup(format!(
+                "Not Merge huge group {} group_size {} next_group {} next_group_size {} size_limit {}",
+                group.group_id, group.group_size, next_group.group_id, next_group.group_size, size_limit
+            )));
+        }
```

Comment:
This check may be inaccurate because we are checking on a potential stale view on `next_group` and `created_tables`. Will this be an issue?

Reply:
Do you means `check_is_creating_compaction_group`? 
This will not cause an error, the version lock is held in merge_compaction_group to check if the table is in the creating state. Consider two scenarios:
1. Since we are using stale view, the new table created after getting the created_tables will not be judged, and it may be misjudged, but merge_compaction_group will intercept this case.
2. the table has already been created and we will try to merge it again in the next scheduling.